addons/portal/static/src/js/portal_signature.js

```diff
@@ -1,93 +1,126 @@
+        },
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
 
-        clearSign: function () {
-            this.$("#o_portal_signature").jSignature('reset');
+        /**
+         * Focuses the name.
+         *
+         * @see NameAndSignature.focusName();
+         */
+        focusName: function () {
+            this.nameAndSignature.focusName();
+        },
+        /**
+         * Resets the signature.
+         *
+         * @see NameAndSignature.resetSignature();
+         */
+        resetSignature: function () {
+            return this.nameAndSignature.resetSignature();
         },
 
-        submitSign: function (ev) {
-            ev.preventDefault();
+        //----------------------------------------------------------------------
+        // Handlers
+        //----------------------------------------------------------------------
 
-            // extract data
+        /**
+         * Handles click on the submit button.
+         *
+         * This will get the current name and signature and validate them.
+         * If they are valid, they are sent to the server, and the reponse is
+         * handled. If they are invalid, it will display the errors to the user.
+         *
+         * @private
+         * @param {Event} ev
+         */
+        _onClickSignSubmit: function (ev) {
             var self = this;
-            var $confirm_btn = self.$el.find('button[type="submit"]');
-
-            // process : display errors, or submit
-            var partner_name = self.$("#o_portal_sign_name").val();
-            var signature = self.$("#o_portal_signature").jSignature('getData', 'image');
-            var is_empty = signature ? this.empty_sign[1] === signature[1] : true;
+            ev.preventDefault();
 
-            this.$('#o_portal_sign_name').parent().toggleClass('o_has_error', !partner_name).find('.form-control, .custom-select').toggleClass('is-invalid', !partner_name);
-            this.$('#o_portal_sign_draw').toggleClass('bg-danger text-white', is_empty);
-            if (is_empty || ! partner_name) {
-                return false;
+            if (!this.nameAndSignature.validateSignature()) {
+                return;
             }
 
-            $confirm_btn.prepend('<i class="fa fa-spinner fa-spin"></i> ');
-            $confirm_btn.attr('disabled', true);
-
-            return rpc.query({
-                route: this.options.callUrl,
-                params: {
-                    'res_id': this.options.resId,
-                    'access_token': this.options.accessToken,
-                    'partner_name': partner_name,
-                    'signature': signature ? signature[1] : false,
-                },
+            var name = this.nameAndSignature.getName();
+            var signature = this.nameAndSignature.getSignatureImage()[1];
+
+            this._rpc({
```

Comment:
BTW why not returning this deferred especially if the handler is async ?

Reply:
Ok that's the problem actually, I suppose the `makeButtonHandler` didn't work because the def was not returned.